docs.feldera.com/docs/pipelines/transactions.md

```diff
@@ -0,0 +1,256 @@
+| **When to use**               | <p>Process real-time inputs with low latency</p>                                                                                               | <p>Efficiently process bulk data</p>                                                                                                                                 |
+| **When query evaluation is triggered** | <p>When an input chunk is ingested</p>                                                                                          | <p>On transaction commit</p>                                                                                                                                         |
+| **Strong consistency**        | <p>Yes: after processing an input chunk, the contents of all views matches all the inputs received so far.</p>                                 | <p>Yes: after committing a transaction the contents of all matches all the inputs received so far, including inputs received as part of the transaction.</p>         |
+
+## Transaction API
+
+### Start a transaction
+
+Use the [`start_transaction`](https://docs.feldera.com/api/start-a-transaction) API to start a transaction. The API returns a transaction ID:
+
+<Tabs>
+    <TabItem value="rest" label="REST API">
+    ```shell
+    $ curl -X POST http://localhost:8080/v0/pipelines/my_pipeline/start_transaction
+
+    {"transaction_id":1}
+    ```
+    </TabItem>
+
+    <TabItem value="python" label="Python SDK">
+    ```python
+    from feldera.pipeline import Pipeline
+
+    transaction_id = pipeline.start_transaction()
+    ```
+    </TabItem>
+
+    <TabItem value="fda" label="fda CLI">
+    ```shell
+    $ fda start-transaction my_pipeline
+
+    Transaction started successfully with ID: 1
+    ```
+    </TabItem>
+</Tabs>
+
+* The `start_transaction` call will fail if there is already a transaction in progress.
+
+* During a transaction, the pipeline ingests incoming data without producing output, performing only minimal processing such as resolving primary keys and indexing inputs.
+
+### Commit a transaction
+
+Once the pipeline has ingested all inputs that must be processed as part of the transaction,
+commit the transaction using the [`commit_transaction`](https://docs.feldera.com/api/commit-the-current-transaction) API.
+
+<Tabs>
+    <TabItem value="rest" label="REST API">
+    ```shell
+    $ curl -X POST http://localhost:8080/v0/pipelines/my_pipeline/commit_transaction
+    "Transaction commit initiated"
+    ```
+    </TabItem>
+
+    <TabItem value="python" label="Python SDK">
+    ```python
+    transaction_id = pipeline.commit_transaction(transaction_id, wait = True)
+    ```
+    </TabItem>
+
+    <TabItem value="fda" label="fda CLI">
+    ```shell
+    $ fda commit-transaction my_pipeline
+
+    Transaction committed successfully.
+    ```
+    </TabItem>
+</Tabs>
+
+* The `commit_transaction` REST API initiates the commit. The commit can take some time and complete later.
```

Comment:
The commit operation will generally require a time proportional to the ingested data size, and may take a significant amount of time when the ingested data is very large.

Is the data processing started only at commit, or is some work done while you push the inputs too?
Or is this irrelevant for users?